### PR TITLE
Clamp cloudwatch values to allowable range

### PR DESCRIPTION
**Pull request by osialr**
"Although the Value parameter accepts numbers of type Double, CloudWatch
rejects values that are either too small or too large. Values must be in
the range of 8.515920e-109 to 1.174271e+108 (Base 10) or 2e-360 to 2e360
(Base 2). In addition, special values (for example, NaN, +Infinity,
-Infinity) are not supported."

http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html

### CODE REVIEW

File: implementations/micrometer-registry-cloudwatch/src/main/java/io/micrometer/cloudwatch/CloudWatchMeterRegistry.java [hunk 1]
```diff
@@ -158,11 +161,11 @@ private MetricDatum metricDatum(Meter.Id id, long wallTime, double value) {
                 .withMetricName(metricName)
                 .withDimensions(toDimensions(tags))
                 .withTimestamp(new Date(wallTime))
-                .withValue(value)
+                .withValue(CloudWatchUtils.clampMetricValue(value))
                 .withUnit(toStandardUnit(id.getBaseUnit()));
     }
 
-    private StandardUnit toStandardUnit(String unit) {
+    private static StandardUnit toStandardUnit(String unit) {
```
Comment (checketts) [inline implementations/micrometer-registry-cloudwatch/src/main/java/io/micrometer/cloudwatch/CloudWatchMeterRegistry.java]: Why are these changed to `static`?

**Comment by osialr** [comment]
SonarLint and Intellij were having a field day in that file and I was trying to clean some of it up.  But I just dropped all of those changes.
